.buildkite/scripts/lib/manifest.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Results in a single sha256sum of all files in directory
+sha256_of_dir() {
+    local -r dir_path="${1}"
+    find "${dir_path}" -type f -exec sha256sum {} \; | sha256sum | awk '{print $1;}'
```

Comment:
I would sort the files before generating the SHA here; running this on my machine yielded output that was definitely _not_ sorted, and since that impacts the final SHA, we should eliminate that source of non-determinism.

```
find "${dir_path}" -type f | sort | xargs sha256sum | sha256sum | awk '{print $1;}'
```

It would also be nice to augment the comment with an example showing exactly what this is doing (you don't need to pick a directory with as many files as this; it's just the directory I was in).

```
# Generates a SHA256 checksum of the sorted output of the SHA256 
# checksums of all the files in this directory (recursively).
#
# $ find . -type f | sort | xargs sha256sum 
# 49a38fd2bd8123074210b621f35abbf8ffe715e5ed764f77fcf0edde70337c5f  ./BUILD
# 503575db67948f54c009463867063fa7cfce651ea2e00c39e81ee8cc4ae69201  ./build_and_upload_firecracker_packages.sh
# 1612b1614be1b0171fa93b89fa703018cf29dd6235be5fce1104836298461fe1  ./build_and_upload_images.sh
# 7d6a4097989400c4b0eb0e9ae0fd6870957e5a136980c6ae2e3d834b20fed103  ./ensure_regenerated_constraints.sh
# 21f84d8a2139892b9cdc755a8f613e85d7d8673b80146dd152fe5c33c2e2f875  ./extract_artifacts.sh
# 49a38fd2bd8123074210b621f35abbf8ffe715e5ed764f77fcf0edde70337c5f  ./grapl_testing_stack/BUILD
# a4431fa56de9407b71138f569e2a5580025e2f1d045a798da37c91cee6b3715e  ./grapl_testing_stack/destroy_stateful_resources.sh
# 0739bc94cee0f49a1a8e389fb2297e44941931ebea71a861ac94289afb44ee53  ./grapl_testing_stack/dump_artifacts.sh
# a96cad5feee9ab749810222daac6c4b34273c6a991477827a0b08e201d57ff73  ./grapl_testing_stack/run_parameterized_job.sh
# 503268de48c7d9ec410e0d98f7171ae4129ab700b1f015fdd5ddf37d8a073ae6  ./lib/artifacts.sh
# 84c15aa349803eb90a2cba09a1325116c68ec6b94397fca75bb69c6260cd1218  ./lib/artifacts_test.sh
# 73ce7319832aec36bec5d844e63a82ae0216e52b83e353f642009d8ec07559bf  ./lib/BUILD
# db37b18d5b9500699077df33b266504933f67131de2ff02dc2cc63e3bd3ea690  ./lib/cloudsmith.sh
# 421ca804d03b5c3ab6aefaa3c4705ec25a61a823fea84ed385d19101a28fb43e  ./lib/lambda.sh
# a9a0d602a7ae3abe48f3b235d1e084d9d08df645b0f088ef5fa270639071b86f  ./lib/lambda_test.sh
# 8364b239db29a18443546a0d1d3d013a32aec8257e64ec2ba511da70caa5b946  ./lib/manifest.sh
# 1c86ebdc5bd4b27d1a3c7d37ca8e40209eb481ead99028ff8712c4f55b4f3da7  ./lib/version.sh
# 777d89e927ae2987f7949df69720fd903523c37250d35eae33b9a3ecd89c598c  ./merge_artifact_files.sh
# 10e6b7d995b77d4b3f4db4066a385cc3b1addb9467b8310f3852bbff389770e5  ./upload_promotion_step_if_needed.sh
# 0f43a2b8e47f36e7218d487907fbb7bfece2b952b4e16fbf8602d1dc3820916f  ./validate_codeowners.sh
#
# Taking the SHA256 checksum of this output yields the final 
# checksum:
#
# $ find . -type f | sort | xargs sha256sum | sha256sum
# 5b611bf839fb19c58f57187053111ff82ebb72cb772a5342b569ec420d39e9e7  -
```